### DISCUSSION TITLE
I'm the maintainer of a large open source project - how can I get CoPilot for free?

### DISCUSSION THREAD

**Discussion by iJungleboy** [community/community#43993] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I'm the maintainer of a large open source project - how can I get CoPilot for free?

It's actually a lot of projects, the most prominent being 2sxc - https://github.com/2sic/2sxc/ - a .net CMS system for DNN / Oqtane used by tens of thousands of websites. 

What must I do to get "verified"?

**Comment by auchynnikau** [comment]
You can visit the GitHub Copilot website and fill out the application program form. The application will ask you to provide information about your open source project, including its size, scope, and impact. Additionally, you'll be asked to provide information about the organization or individual who is maintaining the project, and to explain why the project would benefit from access to GitHub's advanced tools. You can also, when submitting an application you can include any information that you think is relevant to your application, like the number of contributors and how many users your project has, it can help the review process.

Note that, The program is available to select open source projects, and it is not guaranteed that every application will be accepted. Also, the program is subject to change, and GitHub reserves the right to modify the terms of the program or discontinue it at any time.
And finally you can contact GitHub support if you have any question regarding the application process.

**Reply by iJungleboy** [reply]
@auchynnikau thanks for this, I understand that we're at the mercy of Github ;)

But I simply cannot find such an application form. Do you have a link?

**Reply by pratikkabade** [reply]
Did you get any link?

**Comment by spenserblack** [comment]
>GitHub Copilot is free to use for ... maintainers of popular open source projects.

source: https://docs.github.com/en/copilot/quickstart

I'm curious exactly what the threshold is to be considered "popular," and if that's documented anywhere.

**Reply by anggrayudi** [reply]
Agree. I don't know how many stars we should collect before applying to the Copilot's open source license.

**Comment by mikemaccana** [comment]
GitHub says: "If you meet the criteria as an open source maintainer, you will be automatically notified when you visit the GitHub Copilot subscription page."

Which seems false, I have one project with 4K stars and have contributions to node.js, I don't see any notifications at https://github.com/settings/billing/summary, all I can do is upgrade.

**Reply by spenserblack** [reply]
That's a lot of stars! I guess you're referring to [`python-docx`](https://github.com/python-openxml/python-docx)? 

I actually do have Copilot for free, I believe because I'm a maintainer (not an owner, just a collaborator) of a pretty popular repository. So I'm pretty sure that it does work at least some times.

Maybe because the [original `python-docx` that you own](https://github.com/mikemaccana/python-docx) is archived? Are you a maintainer of the new `python-docx`? AFAIK GitHub's definition of a "maintainer" is an owner or collaborator. Maybe an organization team member, too, but I'm not sure. But I don't think "contributor" is enough to be considered a maintainer.

Please note that I'm just guessing here based on observations. I don't have any actual knowledge of the rules.

**Reply by egelhaus** [reply]
I have over 20k but still not getting any notification!

**Comment by mattt-arnolddd** [comment]
That's awesome you're maintaining such a widely used project  (https://github.com/2sic/2sxc).

GitHub offers Copilot for free to verified maintainers of popular open source projects, and your work definitely sounds like a strong candidate.

The verification process isn't publicly documented, but reaching out to GitHub support might be the best way to inquire (https://docs.github.com/support/contacting-github-support). They can give you specific details!

**Reply by iJungleboy** [reply]
This is tricky. I can't open a support ticket without having a paid account. That kind of defeats the purpose - it says any free accounts must use discussions :(

**Comment by 99991** [comment]
https://github.com/pricing#i-work-on-open-source-projects-can-i-get-access-to-github-copilot-for-free

> # I work on open source projects, can I get access to GitHub Copilot for free?

> People who maintain popular open source projects receive a credit to have 12 months of GitHub Copilot access for free. A maintainer of a popular open source project is defined as someone who has write or admin access to one or more of the most popular open source projects on GitHub. Simply visit the GitHub Copilot subscription page to see if you are one of the open source maintainers that meet our criteria for a complimentary subscription. If you do, you should see that you can add GitHub Copilot for no charge. If you see a charge on the purchase page then this means that you do not qualify at this time. Once awarded, if you are still a maintainer of a popular open source project when your initial 12 months subscription expires then you will be able to renew your subscription for free.

In short, you should be eligible if you have write access to a "popular open source project" on GitHub. [GitHub employee](https://npf.io/about/) @natefinch [says](https://news.ycombinator.com/item?id=31826923) that "popular open source projects" must have a minimum number of stars and forks:

> There's a definition somewhere in the FAQ, it's like the top 1000 projects in each of the top 34 most popular languages on GitHub, as long as those projects have some minimum number of stars and forks.

For reference, I maintain [PyMatting](https://github.com/pymatting/pymatting) with 1754 stars and 216 forks at time of writing and am not eligible.

**Reply by TryTryAgain** [reply]
So it has to be _one of the most popular on all of github_ ... that's very different than being a maintainer of a popular open source project.

**Reply by oxisto** [reply]
It is not 100 % transparent for me. I recently got offered the pro version for free. I am mainly the co-maintainer of https://github.com/golang-jwt/jwt. We have about 7-8k stars, which is definitely not little, but not sure if it qualifies as most popular of all go language projects. We do however have quite a large "consumer" base, depending on the version 30.000-50.000 repositories depend on it, so maybe they are factoring in dependency data as well.

**Comment by StefanieSenger** [comment]
I just went through the process. It was a bit confusing, but worked out in the end. [These docs](https://docs.github.com/en/copilot/how-tos/manage-your-account/get-free-access-to-copilot-pro) suggest that you can see in your copilot settings if you are eligible for "GitHub Copilot Pro" due to being an open source maintainer (quote "you will see a page titled "GitHub Copilot Pro" informing you that you are eligible"). This was however not the case of me. 

I then decided to subscribe to the paid Pro version and only once I had inserted all my credentials and send the subscription, I got directed to a page that said "Congratulations. You are eligible to use Copilot for free" and informing me that they check if requirements are still fulfilled monthly. :tada: 

(I'm a maintainer on a project that has 64k stars.)

**Reply by martibosch** [reply]
Hello! thanks for sharing this, I think this is also my case. Does this mean that you had to add a payment method, or the "Congratulations" message appeared before that? Thanks!

**Reply by StefanieSenger** [reply]
I indeed had to add a payment method and go through all the steps to actually submit a (possible) payment. And only then I go to see that I am eligible for Copilot for free...

**Reply by mendhak** [reply]
Can confirm, it feels like a bit of a 'roulette'. I had to go through all the way, submitting billing info, credit card info, and then clicking 'Activate'. So at this point they could have actually charged me but then I got the page saying I was eligible.

**Comment by greatmengqi** [comment]
Hi, I'm a core committer of [bytedance/deer-flow](https://github.com/bytedance/deer-flow) (43k+ GitHub stars, 5k+ forks, active development).                                                                                                                    
  I'd like to request free Copilot Pro (or Pro+ if available) access as an open-source maintainer.                                                                                                                       
  My GitHub handle is @greatmengqi.

**Comment by EvaMarco** [comment]
Hi, I'm a core contributor to [penpot](https://github.com/penpot/penpot), a widely used open-source design platform with a large and active community and continuous development.

I have been contributing to the project for over six years and I am currently the 5th top contributor in terms of commits, actively maintaining and improving the codebase.

Given my long-term commitment and impact on a significant open-source project, I'd like to request complimentary access to Copilot Pro (or Pro+ if available) as an open-source maintainer. I believe it would meaningfully enhance my productivity and allow me to contribute even more effectively.

My GitHub handle is @EvaMarco.

Thank you for your time and consideration.

**Comment by iJungleboy** [comment]
I will close this now, posting here has no effect.
